coingecko/coingecko-fetcher.ts

```diff
@@ -1,123 +0,0 @@
-import { kv } from "@vercel/kv";
-import moment from "moment";
-import { userKey } from "./constants";
-import { coingeckoAPIErrorResponse, invalidResponseTypesError } from "./errors";
-import { PriceDataResponse } from "./interfaces";
-import { coinList } from "./supported-coins";
-import {
-  CoingeckoResponse,
-  getCoingeckoRangeURL,
-  getDayId,
-  getTimestampFromDayId,
-  KVDataToPrice,
-  parsePriceResponse,
-} from "./utils";
-
-const fetchData = async (symbol: string, start: number, finish: number) => {
-  const id = coinList[symbol];
-  const url = getCoingeckoRangeURL(id, start, finish);
-  const res = await fetch(
-    `https://api.scraperapi.com/?api_key=${
-      process.env.SCRAPER_API_KEY
-    }&url=${encodeURIComponent(url)}`
-  );
-
-  if (res.status !== 200) {
-    throw coingeckoAPIErrorResponse(res);
```

Comment:
What if we shield our internal messages from the end user's eyes? How about logging an error for the admin's eyes only and throwing "Internal error code X" kind of message to the client?

Reply:
do you want me to add the same return here? something like that?
```
{
throw coingeckoAPIErrorResponse;
return coingeckoAPIErrorResponse
}

Reply:
```diff
if (res.status !== 200) {
-    throw coingeckoAPIErrorResponse(res);
+    throw console.error(res);
+    throw coingeckoAPIErrorResponse;
```